### DISCUSSION TITLE
Can not connect SignClient on first request

### DISCUSSION THREAD

**Discussion by CaptainHoangLe** [WalletConnect/walletconnect-monorepo#4086] [Developer Support] [answered]
i am using SignClient version  ^2.9.0. On the first time i request connect , it allways pending forever but it allways connect successfully after that. i don 't know where i wronged. Please anyone help me 

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/53038465/0068944e-0745-44b6-acdf-ce32375032b7)

**Comment by glitch-txs** [comment]
hi, which wallet are you using?

**Reply by CaptainHoangLe** [reply]
Leap and Coin98. However , i think bug is missing await in SignClient.init() function of  @walletconnect/sign-client. 
I have fixed by add a promise timeout after init like this.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/53038465/c1dcdd1e-8a69-479f-bb8d-453c5c0d530b)

I read code in https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/sign-client/src/client.ts. I think initialize() is missing await in step "this.core.verify.init({ verifyUrl: this.metadata.verifyUrl });". Hope it is useful. Thanks a lot
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/53038465/1cfefee0-ebac-4c1b-bbfe-e65a7c4888ae)

**Reply by ganchoradkov** [reply, marked as answer]
hey, please update to signClient latest - current is `2.11.0`, tons of fixes and improvements were published from `2.9x`
This should resolve your problem

**Comment by paritaVhits** [comment]
Do you know to register a our own wallet in web3 connect ?
